ext_mongodb.php

```diff
@@ -853,21 +1100,51 @@ public function __debugInfo() : array
 
 final class UTCDateTime implements Type, \Serializable
 {
-	use DenySerialization;
-
 	private int $milliseconds;
 
+	static private function checkArray( array $state )
+	{
+		if (
+			!array_key_exists( 'milliseconds', $state ) ||
+			! ( is_int( $state['milliseconds'] ) || is_string( $state['milliseconds'] ) )
+		) {
+			throw new \MongoDB\Driver\Exception\InvalidArgumentException( "MongoDB\BSON\UTCDateTime initialization requires \"milliseconds\" integer or numeric string field" );
+		}
+	}
+
+	public function serialize() : string
+	{
+		return serialize( [
+			'milliseconds' => (string) $this->milliseconds,
+		] );
+	}
+
+	public function unserialize(mixed $serialized) : void
+	{
+		$unserialized = unserialize( $serialized );
+		self::checkArray( $unserialized );
+		$this->__construct( $unserialized['milliseconds'] );
+	}
+
 	public function __construct(mixed $milliseconds = NULL)
 	{
 		if ($milliseconds === NULL) {
 			$this->milliseconds = floor( microtime( true ) * 1000 );
-		} elseif (is_object($milliseconds) && $milliseconds instanceof \DateTimeInterface) {
+		} elseif (is_object( $milliseconds) && $milliseconds instanceof \DateTimeInterface) {
 			$this->milliseconds = floor( (string) $milliseconds->format('U.u') * 1000 );
-		} else {
+		} elseif ( is_numeric( $milliseconds ) ) {
```

Comment:
On a pedantic note, PHPC uses `atoi()` or `strtoll()` and attempts to verify that the entire string is parsed with no leftover characters. `is_numeric()` is going to allow floating point values before you cast to an integer. [`ctype_digit`](http://php.net/manual/en/function.ctype-digit.php) with an allowance for a leading negation operator might be more consistent.

I don't imagine this being a problem in practice, but expect that a future error test in PHPC could inadvertently pass in HHVM. I won't press for a change here, but I did want to make a note of it.